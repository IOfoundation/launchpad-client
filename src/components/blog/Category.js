import React from 'react';
import {Link} from 'react-router';
import {PropTypes} from 'prop-types';

const Category = props => {
  const {to, label, className} = props;

  return (
    <li className={['post-category', className].join(' ')}>
      <Link to={to}>{label}</Link>
    </li>
  );
};

Category.propTypes = {
  className: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  to: PropTypes.string.isRequired,
};

export default Category;
