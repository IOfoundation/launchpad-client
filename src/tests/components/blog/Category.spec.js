import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import Category from '../../../components/blog/Category';

const clicked = jest.fn();
const props = {
  className: 'testing-class',
  clicked,
  isSelected: false,
  label: 'Testing Label',
};

describe('<Category />', () => {
  it('renders snapshot of Category', () => {
    const wrapper = shallow(<Category {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should execute clicked function', () => {
    const wrapper = shallow(<Category {...props} />);
    const component = wrapper.find('.testing-class');

    component.simulate('click');

    expect(clicked.mock.calls.length).toEqual(1);
  });
});
