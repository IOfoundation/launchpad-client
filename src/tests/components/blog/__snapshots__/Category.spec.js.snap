// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Category /> renders snapshot of Category 1`] = `
<li
  className="blog-categories capitalize testing-class"
  onClick={[MockFunction]}
>
  Testing Label
</li>
`;
